todo/src/TodoList.js

```diff
@@ -0,0 +1,42 @@
+import React, { Component } from 'react';
+import Todo from './Todo.js'
+import { connect } from 'react-redux'
+import { toggleCompletedAction } from './actions'
+import { deleteTodoAction } from './actions'
+
+class TodoList extends Component {
+    clickHandler = (e, payload) => {
+        e.preventDefault();
+        this.props.toggleCompletedAction(payload);
+    }
+
+    deleteClickHandler = (e, payload) => {
+        e.preventDefault();
+        this.props.deleteTodoAction(payload)
+    }
+
+    render() {
+        return (
+            <div>
+                {this.props.todos.map(todo =>
+                    <div>
+                        <Todo 
+                            todo={todo}
+                            key={Math.random()}
+                            onClick={(e) => this.clickHandler(e, todo)}
+                        />
+                        <button onClick={(e) => this.deleteClickHandler(e, todo)}>Delete to-do</button>
+                    </div>
+                )}
+            </div>
+        );
+    }
+}
+
+const mapStateToProps = (state) => {
+    return {
+        todos: state.todos
+    };
+};
+
+export default connect(mapStateToProps, {toggleCompletedAction, deleteTodoAction})(TodoList);
```

Comment:
✏️ Basics Refresher: `connect()` allows us to specify which data we are listening to (through mapStateToProps), and which component we are providing the data.